### PR TITLE
Make loadBalancerIP in ingress-proxy service configurable

### PR DESCRIPTION
**Pull request by maxbischoff**
# Description

This new feature can be used to set the `loadBalancerIP` in the `ingress-proxy` service when installing Gloo as ingress-proxy using helm. Fixes https://github.com/solo-io/gloo/issues/3184

# Context

Users needed this feature to assign static load balancer IPs to their gloo ingress.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make update-deps generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo [contribution guide](https://docs.solo.io/gloo/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/3184

### CODE REVIEW

**Comment by maxbischoff** [comment]
Is a test for this PR required?

**Comment by maxbischoff** [comment]
Small feedback: I had a hard time figuring out, what changelog folder I needed to add my file to. The bot only told me which directory it expected me to add it to, after I had created the wrong folder. Might only be a temporary problem due to the incoming 1.4 release, but it would be nice if the bot would immediately tell me where to add my changelog.

File: changelog/v1.4.0-beta16/make-loadbalancerip-in-ingress-proxy-configurable.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changelog:
+  - type: NEW_FEATURE
```
Comment (kdorosh) [inline changelog/v1.4.0-beta16/make-loadbalancerip-in-ingress-proxy-configurable.yaml:2]: we actually have a changelog type `HELM` for helm changes

Comment (maxbischoff) [reply changelog/v1.4.0-beta16/make-loadbalancerip-in-ingress-proxy-configurable.yaml]: Sorry, I fixed that now.

**Comment by kdorosh** [comment]
looks like a helm test failure
```
Step #8 - "test": • Failure [0.199 seconds]
Step #8 - "test": Helm Test Helm 2 ingress-proxy service [It] sets loadBalancerIP 
Step #8 - "test": /workspace/gloo/install/test/helm_test.go:2047
Step #8 - "test": 
Step #8 - "test":   can't find object Service gloo-system ingress-proxy
Step #8 - "test": 
Step #8 - "test":   /go/pkg/mod/github.com/solo-io/go-utils@v0.14.1/manifesttestutils/util.go:296
Step #8 - "test": 
Step #8 - "test":   Full Stack Trace
Step #8 - "test":   github.com/solo-io/go-utils/manifesttestutils.(*testManifest).mustFindObject(0xc000b6a980, 0x302d475, 0x7, 0x30355b3, 0xb, 0x3039555, 0xd, 0xc000b6a980, 0x0)
Step #8 - "test":   	/go/pkg/mod/github.com/solo-io/go-utils@v0.14.1/manifesttestutils/util.go:296 +0x205
Step #8 - "test":   github.com/solo-io/go-utils/manifesttestutils.(*testManifest).ExpectService(0xc000b6a980, 0xc0005ccb40)
Step #8 - "test":   	/go/pkg/mod/github.com/solo-io/go-utils@v0.14.1/manifesttestutils/util.go:156 +0xff
Step #8 - "test":   github.com/solo-io/gloo/install/test.glob..func2.2.1.10.4()
Step #8 - "test":   	/workspace/gloo/install/test/helm_test.go:2054 +0x225
Step #8 - "test":   github.com/onsi/ginkgo/internal/leafnodes.(*runner).runSync(0xc0003eed80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
Step #8 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/leafnodes/runner.go:113 +0xe0
Step #8 - "test":   github.com/onsi/ginkgo/internal/leafnodes.(*runner).run(0xc0003eed80, 0xc000343128, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
Step #8 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/leafnodes/runner.go:64 +0x108
Step #8 - "test":   github.com/onsi/ginkgo/internal/leafnodes.(*ItNode).Run(0xc0005ecf00, 0x3577a20, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
Step #8 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/leafnodes/it_node.go:26 +0x88
Step #8 - "test":   github.com/onsi/ginkgo/internal/spec.(*Spec).runSample(0xc0006dea50, 0x0, 0x3577a20, 0xc000343100)
Step #8 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/spec/spec.go:215 +0x6e2
Step #8 - "test":   github.com/onsi/ginkgo/internal/spec.(*Spec).Run(0xc0006dea50, 0x3577a20, 0xc000343100)
Step #8 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/spec/spec.go:138 +0x168
Step #8 - "test":   github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).runSpec(0xc0003d0140, 0xc0006dea50, 0x0)
Step #8 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/specrunner/spec_runner.go:200 +0x173
Step #8 - "test":   github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).runSpecs(0xc0003d0140, 0x1)
Step #8 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/specrunner/spec_runner.go:170 +0x217
Step #8 - "test":   github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).Run(0xc0003d0140, 0x8)
Step #8 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/specrunner/spec_runner.go:66 +0x141
Step #8 - "test":   github.com/onsi/ginkgo/internal/suite.(*Suite).Run(0xc00032d130, 0x7f9510534750, 0xc0001c2e00, 0x3032a68, 0xa, 0xc00046cbb0, 0x1, 0x1, 0x35f8fc0, 0xc000343100, ...)
Step #8 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/suite/suite.go:62 +0x6a0
Step #8 - "test":   github.com/onsi/ginkgo.RunSpecsWithCustomReporters(0x35798c0, 0xc0001c2e00, 0x3032a68, 0xa, 0xc0000cdea0, 0x1, 0x1, 0xc000014070)
Step #8 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/ginkgo_dsl.go:226 +0x339
Step #8 - "test":   github.com/onsi/ginkgo.RunSpecs(0x35798c0, 0xc0001c2e00, 0x3032a68, 0xa, 0x1)
Step #8 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/ginkgo_dsl.go:207 +0x24e
Step #8 - "test":   github.com/solo-io/gloo/install/test.TestHelm(0xc0001c2e00)
Step #8 - "test":   	/workspace/gloo/install/test/helm_suite_test.go:45 +0x256
Step #8 - "test":   testing.tRunner(0xc0001c2e00, 0x3133610)
Step #8 - "test":   	/usr/local/go/src/testing/testing.go:909 +0x19a
Step #8 - "test":   created by testing.(*T).Run
Step #8 - "test":   	/usr/local/go/src/testing/testing.go:960 +0x652
Step #8 - "test": ------------------------------
```

**Comment by maxbischoff** [comment]
> looks like a helm test failure

Should be fixed now

**Pull request merged**